src/Inventory/Asset/Unmanaged.php

```diff
@@ -0,0 +1,390 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Inventory\Asset;
+
+use AutoUpdateSystem;
+use Glpi\Inventory\Asset\Printer as AssetPrinter;
+use Glpi\Inventory\Conf;
+use Glpi\Inventory\Request;
+use Glpi\Toolbox\Sanitizer;
+use NetworkEquipment;
+use NetworkName;
+use NetworkPortInstantiation;
+use Printer;
+use RefusedEquipment;
+use RuleMatchedLog;
+use Toolbox;
+use Transfer;
+
+class Unmanaged extends MainAsset
+{
+    private $management_ports = [];
+
+    protected $extra_data = [
+        'hardware'        => null,
+        'network_device'  => null,
+    ];
+
+    public function prepare(): array
+    {
+        parent::prepare();
+        $val = $this->data[0];
+
+        $raw_data = $this->raw_data;
+        if (!is_array($raw_data)) {
+            $raw_data = [$raw_data];
+        }
+
```

Comment:
Code is useless.
```suggestion
```

Reply:
I don't think so, this is exactly what the ```MainAsset``` class does when ```prepare```